### PR TITLE
Additional existingBinding Injector extension

### PR DESCRIPTION
**Pull request by jw3**
This one allows the Key to be computed externally.

### CODE REVIEW

File: src/main/scala/net/codingwell/scalaguice/InjectorExtensions.scala [hunk 1]
```diff
@@ -26,8 +26,9 @@ object InjectorExtensions {
     def instance[T: Manifest](ann: Annotation) = i.getInstance(typeLiteral[T].annotatedWith(ann))
     def instance[T: Manifest, Ann <: Annotation : Manifest] = i.getInstance(typeLiteral[T].annotatedWith[Ann])
 
-    def existingBinding[T: Manifest] = Option(i.getExistingBinding(typeLiteral[T].toKey))
-    def existingBinding[T: Manifest](ann: Annotation) = Option(i.getExistingBinding(typeLiteral[T].annotatedWith(ann)))
-    def existingBinding[T: Manifest, Ann <: Annotation : Manifest] = Option(i.getExistingBinding(typeLiteral[T].annotatedWith[Ann]))
+    def existingBinding[T: Manifest]: Option[Binding[T]] = existingBinding(typeLiteral[T].toKey)
+    def existingBinding[T: Manifest](ann: Annotation): Option[Binding[T]] = existingBinding(typeLiteral[T].annotatedWith(ann))
+    def existingBinding[T: Manifest, Ann <: Annotation : Manifest]: Option[Binding[T]] = existingBinding(typeLiteral[T].annotatedWith[Ann])
+    def existingBinding[T: Manifest](key: => Key[T]): Option[Binding[T]] = Option(i.getExistingBinding(key))
```
Comment (nbauernfeind) [inline src/main/scala/net/codingwell/scalaguice/InjectorExtensions.scala]: Why pass by reference here? I think you just want `(key: Key[T])`.

Comment (jw3) [inline src/main/scala/net/codingwell/scalaguice/InjectorExtensions.scala]: It must have seemed sane at the time, looking at it again I concur with by-value.  

Ill update it.

**Comment by nbauernfeind** [comment]
This also LGTM. I didn't notice before, but you could also drop the Manifest requirement on the Key-based method if you like.

**Comment by jw3** [comment]
>you could also drop the Manifest requirement on the Key-based method if you like

Absolutely, good catch.

**Comment by tsuckow** [comment]
Published a new snapshot. Sorry for the delay

**Comment by jw3** [comment]
Thanks, I'll give it a run.

**Comment by jw3** [comment]
Looks good here

**Pull request merged**